topic-operator/src/main/java/io/strimzi/operator/topic/TopicSerialization.java

```diff
@@ -45,34 +49,30 @@
     public static final String JSON_KEY_REPLICAS = "replicas";
     public static final String JSON_KEY_CONFIG = "config";
 
+    private static void validateConfiguration(Reconciliation reconciliation, KafkaTopic kafkaTopic, String desiredVersion, KafkaTopicConfiguration configuration) {
+        List<String> errorsInConfig = configuration.validate(desiredVersion);
+
+        if (!errorsInConfig.isEmpty()) {
+            for (String error : errorsInConfig) {
+                LOGGER.warnCr(reconciliation, "KafkaTopic {}/{} has invalid spec.config: {}",
+                        kafkaTopic.getMetadata().getNamespace(),
+                        kafkaTopic.getMetadata().getName(),
+                        error);
+            }
+
+            throw new InvalidResourceException("KafkaTopic " +
+                    kafkaTopic.getMetadata().getNamespace() + "/" + kafkaTopic.getMetadata().getName() +
+                    " has invalid spec.config: " +
+                    String.join(", ", errorsInConfig));
+        }
+    }
+
     @SuppressWarnings("unchecked")
-    private static Map<String, String> topicConfigFromTopicConfig(KafkaTopic kafkaTopic) {
+    private static Map<String, String> topicConfigFromTopicConfig(KafkaTopic kafkaTopic, Reconciliation reconciliation, String kafkaVersion) {
         if (kafkaTopic.getSpec().getConfig() != null) {
-            Map<String, String> result = new HashMap<>(kafkaTopic.getSpec().getConfig().size());
-            for (Map.Entry<String, Object> entry : kafkaTopic.getSpec().getConfig().entrySet()) {
-                String key = entry.getKey();
-                Object v = entry.getValue();
-                boolean isNumberType = v instanceof Long
-                        || v instanceof Integer
-                        || v instanceof Short
-                        || v instanceof Double
-                        || v instanceof Float;
-                if (v instanceof String
-                        || isNumberType
-                        || v instanceof Boolean) {
-                    result.put(key, v.toString());
-                } else {
-                    String msg = "The value corresponding to the key must have a string, number or boolean value";
-                    if (v == null) {
-                        msg += " but the value was null";
-                    } else {
-                        msg += " but was of type " + v.getClass().getName();
-                    }
-                    throw new InvalidTopicException(kafkaTopic, "KafkaTopic's spec.config has invalid entry: " +
-                            "The key '" + key + "' of the topic config is invalid: " + msg);
-                }
-            }
-            return result;
+            KafkaTopicConfiguration cfg = new KafkaTopicConfiguration(reconciliation, kafkaTopic.getSpec().getConfig().entrySet());
+            validateConfiguration(reconciliation, kafkaTopic, kafkaVersion, cfg);
```

Comment:
@tombentley Can you see a better place to perform the validation?

Reply:
I just wanted to be consistent with the broker (and eventually connect) config validation.

Reply:
What do others think? Should we keep the validation of the Topic configuration as simple as possible or inherit it from the broker validation? @strimzi/maintainers